### PR TITLE
Analytics dashboard

### LINKED ISSUE #88: Analytics Dashboard
The Internship details made earlier were:
 |Sr No | Company Name | Industry | Location | No-of offers(as per JAF) | Open For/Eligible | Company POC Name | Designation | Contact | Student POC Name | Stipend p/m | Dates Confirmed forOA/PPT/Interviews | Job Profile/JD | No. of offers made | No of PPO conversion | CTC Offered|
|-------|--------------|----------|----------|---------------------------|--------------------|------------------|-------------|---------|------------------|-------------|-------------------------------------|----------------|---------------------|------------------------|-------------|

We need the above fields for FTE: 
|Sr No | Company Name | Industry | Company Location | No-of offers(as per JAF) | Open For/Eligible | Company POC Name | Designation | Contact | Student POC Name | CTC Details | Dates Confirmed forOA/PPT/Interviews | Job Profile/JD | No. of offers made | CTC Offered | D.O.J | Job Location |
|-------|--------------|----------|------------------|---------------------------|--------------------|------------------|-------------|---------|------------------|-------------|--------------------------------------|----------------|---------------------|------------------------|-------|--------------|

For charts and graphs waiting for credentials of sample POD from sir.

### PR DESCRIPTION
**Pull request by AryanGKulkarni**
Closes #88 

Following data will be returned for a particular seasonId

- totalRegisteredStudentsCount
- placedStudentsCount
- placementPercentage
- unplacedPercentage
- totalOffers
- totalCompaniesOffering
- highestPackage
- lowestPackage
- meanPackage
- medianPackage
- modePackage

This Data can be accessed

- Overall
- Course Wise
- Department Wise
- Category Wise
- Gender Wise

### CODE REVIEW

File: src/analytics-dashboard/analytics-dashboard.controller.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+@UseGuards(AuthGuard("jwt"))
+@ApiTags("analytics-dashboard")
+@ApiBearerAuth("jwt")
+export class AnalyticsDashboardController {
+  constructor(private readonly analyticsDashboardService: AnalyticsDashboardService) {}
+
+  @Get("statsOverall/:seasonId")
+  @ApiResponse({ type: StatsDto })
+  async getPlacementStatisticsOverall(@Param("seasonId") seasonId: string, @User() user: IUser) {
+    if (user.role !== "ADMIN") {
```
Comment (mittal-ishaan) [inline src/analytics-dashboard/analytics-dashboard.controller.ts:25]: it would be better if we have another auth guard for admin and use it here.

Comment (AryanGKulkarni) [reply src/analytics-dashboard/analytics-dashboard.controller.ts:25]: another auth guard?

File: src/analytics-dashboard/analytics-dashboard.service.ts [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+      };
+    }
+
+    const highestPackage = Math.max(...ctcArray);
+    const lowestPackage = Math.min(...ctcArray);
+    const meanPackage = ctcArray.reduce((acc, val) => acc + val, 0) / ctcArray.length;
+
+    ctcArray.sort((a, b) => a - b);
+    const mid = Math.floor(ctcArray.length / 2);
+    const medianPackage = ctcArray.length % 2 !== 0 ? ctcArray[mid] : (ctcArray[mid - 1] + ctcArray[mid]) / 2;
```
Comment (mittal-ishaan) [inline src/analytics-dashboard/analytics-dashboard.service.ts:42]: we can maybe use https://mathjs.org/docs/reference/functions/median.html for all these. only a suggestion, but your call

File: src/analytics-dashboard/analytics-dashboard.controller.ts (hunk 1 above)
Comment (mittal-ishaan) [reply src/analytics-dashboard/analytics-dashboard.controller.ts:25]: yes

Comment (mittal-ishaan) [reply src/analytics-dashboard/analytics-dashboard.controller.ts:25]: like we have for jwt we can have another that extends it and also checks for role== admin

**Pull request merged**